### ISSUE TITLE
Authentication with Google, facebook, Twitter  accounts

### ISSUE THREAD

**Issue by ducmaxime** [delight-im/Android-DDP#21]
Hello,

it's possible connect to meteor server with Google or Facebook  account ?

tks

**Comment by mwaclawek** [comment]
Thanks for your question!

Unfortunately, this is not yet supported by this library. Sorry! Meteor itself (for the web) [supports this](http://docs.meteor.com/#/full/meteor_loginwithexternalservice), of course.

Having access to the `Meteor#call(...)` method, however, you could probably implement this yourself, if you'd like:

To connect with the third-party service (e.g. Facebook), one must (most likely) make a separate HTTP request to get a login token. After that, you can then use that token in some method call to your Meteor server using `Meteor#call(...)`.

Hope that helps!

**Comment by cpagr** [comment]
Hi, this is now possible with the new pull request I have made. Kindly check here, https://github.com/cprakashagr/Android-DDP/tree/GoogleLoginHandler

**Comment by cprakashagr** [comment]
Hi, this is now possible with the new pull request I have made. Kindly check here, https://github.com/cprakashagr/Android-DDP/tree/GoogleLoginHandler

**Comment by najeebullahshah** [comment]
@cprakashagr can you please refer to the method which will connect to meteor server via facebook login

**Comment by cprakashagr** [comment]
@najeebullahshah there is no such custom methods for facebook or google login at meteor accounts. Kindly see my code and use the facebook login accordingly.

**Comment by najeebullahshah** [comment]
@cprakashagr  if i am not wrong you added method "registerWithGoogle" to the DDP settings parameters using Map but it uses the following paramters

                accountData.put("email", email);
		accountData.put("userId", userId);
		accountData.put("idToken", idToken);
		accountData.put("oAuthToken", oAuthToken);
		accountData.put("googleLoginPlugin", googleLoginPlugin);

Can you please tell me what parameters should i change for facebook.

**Comment by cprakashagr** [comment]
@najeebullahshah Man, I haven't worked with the facebook oAuth API as of now. This link might help you, https://developers.facebook.com/docs/facebook-login/android#get_current.

**Comment by cprakashagr** [comment]
Kindly be updated about the LinkedIn Login as well.

**Comment by mwaclawek** [comment]
For Google and LinkedIn, please take a look at @cprakashagr's code here: https://github.com/delight-im/Android-DDP/pull/40

For Facebook and Twitter, someone would have to write something similar that consists of ...

 * server-side code in JavaScript for use with Meteor (the bigger part)
 * client-side code in Java/Android for use with this library (the smaller part)

**Comment by gopinath-langote** [comment]
Hello Everyone,
I have implemented loginWithGmail functionality to Meteor DDP for my project. Using server side method. By loginWithUsername or registerWithUser name we should have password. but for loginWithGmail we don't need password. So I published method called createUser on server side and allows client to call it [ Specifically mobile client android / IOS ]. this method return me the document of created user. For sign in with google , i have user [GoogleAuth](https://developers.google.com/identity/sign-in/android/start-integrating) by goole. One can ping for more details and implementation stuff. If one could allow me to insert code in existing library of Meteor class . We can go ahead.
